pkg/ext-proc/backend/provider.go

```diff
@@ -120,56 +151,13 @@ func (p *Provider) refreshPodsOnce() {
 		pod := k.(Pod)
 		if _, ok := p.datastore.pods.Load(pod); !ok {
 			p.podMetrics.Delete(pod)
```

Comment:
Should we remove this, given the LoadAndDelete below?

Reply:
The map loads the pod is different from the map deletes the pod.

Reply:
I am confused, in line 161 `p.podMetrics.Delete(pod)`, and then in line 162, `p.podMetrics.LoadAndDelete(pod)`, won't line 162 always return false since the pod is already deleted in line 161?

Reply:
Thanks for catching up! Sorry for misunderstanding you at the beginning. I will remove the `p.podMetrics.Delete(pod)` soon.

Reply:
Why do we need another map in the first place? can't we put the PodMetricsRefresher in the PodMetrics struct?

Reply:
@ahg-g 
I tried it before. But if we put PodMetricsRefresher under PodMetrics, it's not easy to make it concurrency-safe without a big refactor. As PodMetricsRefresher needs to change PodMetrics on the fly.

Reply:
I believe we should be able to do that, I sent out https://github.com/kubernetes-sigs/gateway-api-inference-extension/pull/350 which does a major refactor to the datastore/provider layer. It consolidating storage in one place under datastore, and so it does in-place updates to the metrics. Please take a look and advice is there are any concurrency issues that I may have missed.